LoopKit/GlucoseKit/StoredGlucoseSample.swift

```diff
@@ -39,7 +40,8 @@ public struct StoredGlucoseSample: GlucoseSampleValue, Equatable {
             trend: sample.trend,
             isDisplayOnly: sample.isDisplayOnly,
             wasUserEntered: sample.wasUserEntered,
-            device: sample.device)
+            device: sample.device,
+            healthKitEligibleDate: sample.startDate)
```

Comment:
If this is from an `HKQuantitySample` (meaning it is already from HealthKit) then shouldn't `healthKitEligibleDate` be `nil`?

Reply:
Good catch.  TBH I didn't know _what_ to put here; the only caller of this `init` is `LoopDataManager.predictGlucoseFromManualGlucose()`.  I originally put `nil` but wasn't sure that was right either, but now that you comment on it I think `nil` is definitely better.